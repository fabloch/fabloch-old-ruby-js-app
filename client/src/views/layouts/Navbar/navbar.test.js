import React from "react"
import { mount } from "enzyme"
import toJson from "enzyme-to-json"
import { BrowserRouter as Router } from "react-router-dom"

import { Navbar } from "./"

const notLoggedIn = {
  data: null,
  isFetching: false,
}

const loggedIn = {
  data: {
    client: "abcdef",
    uid: "ghijkl",
    token: "mnopqr",
    expiry: "123456",
  },
  isFetching: false,
}

const logout = jest.fn()

describe("Navbar presentational", () => {
  describe("when not logged in", () => {
    it("homepage", () => {
      const wrapper = mount(
        <Router>
          <Navbar
            pathname={"/"}
            session={notLoggedIn}
            logout={logout}
          />
        </Router>,
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it("signup page", () => {
      const wrapper = mount(
        <Router>
          <Navbar
            pathname={"/account/signup"}
            session={notLoggedIn}
            logout={logout}
          />
        </Router>,
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it("login page", () => {
      const wrapper = mount(
        <Router>
          <Navbar
            pathname={"/account/login"}
            session={notLoggedIn}
            logout={logout}
          />
        </Router>,
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe("when logged in", () => {
    it("homepage", () => {
      const wrapper = mount(
        <Router>
          <Navbar
            pathname={"/"}
            session={loggedIn}
            logout={logout}
          />
        </Router>,
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})
