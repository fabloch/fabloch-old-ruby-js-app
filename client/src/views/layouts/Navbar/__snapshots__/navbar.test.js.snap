exports[`Navbar presentational when logged in homepage 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }>
    <Component
      logout={[Function]}
      pathname="/"
      session={
        Object {
          "data": Object {
            "client": "abcdef",
            "expiry": "123456",
            "token": "mnopqr",
            "uid": "ghijkl",
          },
          "isFetching": false,
        }
      }>
      <Menu>
        <div
          className="ui menu">
          <LeftMenu>
            <MenuMenu>
              <div
                className="menu">
                <MenuItem
                  activeOnlyWhenExact={true}
                  label="La FABrique du Loch"
                  to="/">
                  <Route
                    exact={true}
                    path="/">
                    <MenuItem
                      active={true}
                      as={[Function]}
                      to="/">
                      <Link
                        className="active item"
                        onClick={[Function]}
                        replace={false}
                        to="/">
                        <a
                          className="active item"
                          href="/"
                          onClick={[Function]}>
                          La FABrique du Loch
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="My Fablab"
                  to="/myfablab">
                  <Route
                    path="/myfablab">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/myfablab">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/myfablab">
                        <a
                          className="item"
                          href="/myfablab"
                          onClick={[Function]}>
                          My Fablab
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </LeftMenu>
          <RightMenu>
            <MenuMenu
              position="right">
              <div
                className="right menu">
                <MenuItem
                  label="Me déconnecter"
                  onClick={[Function]}
                  to="/">
                  <Route
                    path="/">
                    <MenuItem
                      active={true}
                      as={[Function]}
                      onClick={[Function]}
                      to="/">
                      <Link
                        className="active item"
                        onClick={[Function]}
                        replace={false}
                        to="/">
                        <a
                          className="active item"
                          href="/"
                          onClick={[Function]}>
                          Me déconnecter
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </RightMenu>
        </div>
      </Menu>
    </Component>
  </Router>
</BrowserRouter>
`;

exports[`Navbar presentational when not logged in homepage 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }>
    <Component
      logout={[Function]}
      pathname="/"
      session={
        Object {
          "data": null,
          "isFetching": false,
        }
      }>
      <Menu>
        <div
          className="ui menu">
          <LeftMenu>
            <MenuMenu>
              <div
                className="menu">
                <MenuItem
                  activeOnlyWhenExact={true}
                  label="La FABrique du Loch"
                  to="/">
                  <Route
                    exact={true}
                    path="/">
                    <MenuItem
                      active={true}
                      as={[Function]}
                      to="/">
                      <Link
                        className="active item"
                        onClick={[Function]}
                        replace={false}
                        to="/">
                        <a
                          className="active item"
                          href="/"
                          onClick={[Function]}>
                          La FABrique du Loch
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="My Fablab"
                  to="/myfablab">
                  <Route
                    path="/myfablab">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/myfablab">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/myfablab">
                        <a
                          className="item"
                          href="/myfablab"
                          onClick={[Function]}>
                          My Fablab
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </LeftMenu>
          <RightMenu>
            <MenuMenu
              position="right">
              <div
                className="right menu">
                <MenuItem
                  label="Me connecter"
                  to="/session/login">
                  <Route
                    path="/session/login">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/session/login">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/session/login">
                        <a
                          className="item"
                          href="/session/login"
                          onClick={[Function]}>
                          Me connecter
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="M\'inscrire"
                  to="/session/signup">
                  <Route
                    path="/session/signup">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/session/signup">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/session/signup">
                        <a
                          className="item"
                          href="/session/signup"
                          onClick={[Function]}>
                          M\'inscrire
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </RightMenu>
        </div>
      </Menu>
    </Component>
  </Router>
</BrowserRouter>
`;

exports[`Navbar presentational when not logged in login page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }>
    <Component
      logout={[Function]}
      pathname="/account/login"
      session={
        Object {
          "data": null,
          "isFetching": false,
        }
      }>
      <Menu>
        <div
          className="ui menu">
          <LeftMenu>
            <MenuMenu>
              <div
                className="menu">
                <MenuItem
                  activeOnlyWhenExact={true}
                  label="La FABrique du Loch"
                  to="/">
                  <Route
                    exact={true}
                    path="/">
                    <MenuItem
                      active={true}
                      as={[Function]}
                      to="/">
                      <Link
                        className="active item"
                        onClick={[Function]}
                        replace={false}
                        to="/">
                        <a
                          className="active item"
                          href="/"
                          onClick={[Function]}>
                          La FABrique du Loch
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="My Fablab"
                  to="/myfablab">
                  <Route
                    path="/myfablab">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/myfablab">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/myfablab">
                        <a
                          className="item"
                          href="/myfablab"
                          onClick={[Function]}>
                          My Fablab
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </LeftMenu>
          <RightMenu>
            <MenuMenu
              position="right">
              <div
                className="right menu">
                <MenuItem
                  label="Me connecter"
                  to="/session/login">
                  <Route
                    path="/session/login">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/session/login">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/session/login">
                        <a
                          className="item"
                          href="/session/login"
                          onClick={[Function]}>
                          Me connecter
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="M\'inscrire"
                  to="/session/signup">
                  <Route
                    path="/session/signup">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/session/signup">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/session/signup">
                        <a
                          className="item"
                          href="/session/signup"
                          onClick={[Function]}>
                          M\'inscrire
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </RightMenu>
        </div>
      </Menu>
    </Component>
  </Router>
</BrowserRouter>
`;

exports[`Navbar presentational when not logged in signup page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }>
    <Component
      logout={[Function]}
      pathname="/account/signup"
      session={
        Object {
          "data": null,
          "isFetching": false,
        }
      }>
      <Menu>
        <div
          className="ui menu">
          <LeftMenu>
            <MenuMenu>
              <div
                className="menu">
                <MenuItem
                  activeOnlyWhenExact={true}
                  label="La FABrique du Loch"
                  to="/">
                  <Route
                    exact={true}
                    path="/">
                    <MenuItem
                      active={true}
                      as={[Function]}
                      to="/">
                      <Link
                        className="active item"
                        onClick={[Function]}
                        replace={false}
                        to="/">
                        <a
                          className="active item"
                          href="/"
                          onClick={[Function]}>
                          La FABrique du Loch
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="My Fablab"
                  to="/myfablab">
                  <Route
                    path="/myfablab">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/myfablab">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/myfablab">
                        <a
                          className="item"
                          href="/myfablab"
                          onClick={[Function]}>
                          My Fablab
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </LeftMenu>
          <RightMenu>
            <MenuMenu
              position="right">
              <div
                className="right menu">
                <MenuItem
                  label="Me connecter"
                  to="/session/login">
                  <Route
                    path="/session/login">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/session/login">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/session/login">
                        <a
                          className="item"
                          href="/session/login"
                          onClick={[Function]}>
                          Me connecter
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
                <MenuItem
                  label="M\'inscrire"
                  to="/session/signup">
                  <Route
                    path="/session/signup">
                    <MenuItem
                      active={null}
                      as={[Function]}
                      to="/session/signup">
                      <Link
                        className="item"
                        onClick={[Function]}
                        replace={false}
                        to="/session/signup">
                        <a
                          className="item"
                          href="/session/signup"
                          onClick={[Function]}>
                          M\'inscrire
                        </a>
                      </Link>
                    </MenuItem>
                  </Route>
                </MenuItem>
              </div>
            </MenuMenu>
          </RightMenu>
        </div>
      </Menu>
    </Component>
  </Router>
</BrowserRouter>
`;
